ui-admin/src/study/participants/enrolleeView/ParticipantNoteView.tsx

```diff
@@ -0,0 +1,37 @@
+import { AdminUser, Enrollee, ParticipantNote } from 'api/api'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faUser } from '@fortawesome/free-solid-svg-icons'
+import { instantToDefaultString } from 'util/timeUtils'
+import { Link } from 'react-router-dom'
+import { enrolleeKitRequestPath } from './EnrolleeView'
+import React from 'react'
+
+
+type ParticipantNoteViewProps = {
+  enrollee: Enrollee,
+  note: ParticipantNote,
+  currentEnvPath: string,
+  users: AdminUser[]
+}
+/** renders a single note, with the admin user name and a link to the kit request, if it exists */
+export function ParticipantNoteView({ enrollee, note, users, currentEnvPath }: ParticipantNoteViewProps) {
+  const matchedUser = users.find(user => user.id === note.creatingAdminUserId)
```

Comment:
yup, that's one valid approach (I know it's what you did with kit types).  The drawback is that it couples the API backend more tightly to the frontend, and can result in a lot of duplicate fetching (e.g. we might need to fetch admin users along with kits, notes, tasks, etc...  So that's why I proposed we move instead to an architecture where there's an initial background load of all the supporting collections (users, kit types, etc...), and then the frontend can decide when it wants to display additional detail about something.  Both work, and we can wait until we've got a couple more cases like this to make a decision

Reply:
I know this is already merged, but just to clarify my thoughts about this...

Resolving server-side does couple the controller (and perhaps service, depending on division of responsibilities) to the UI, but controllers exist to serve interfaces so I don't think that's inherently bad. Besides, resolving client-side couples the UI to the database model. Pre-fetching can also be inefficient (the client might fetch data it never needs), and it feels to me a bit like premature optimization. In general, I'm more worried about identifying and fixing subtle bugs with cached, non-refreshed state in the client than I am in finding ways to optimize queries when we need to improve performance.